nucliadb_vectors/src/formula/mod.rs

```diff
@@ -74,6 +77,14 @@ impl AtomClause {
 
                 field_set.contains(std::str::from_utf8(&key[0..end_pos]).unwrap())
             }
+            Self::KeyField((field_type, field_name)) => {
+                let key = retriever.get_key(x);
+                let mut key_parts = key.split(|b| *b == b'/');
+                key_parts.next(); // Skip resource_id
+                let ftype = std::str::from_utf8(key_parts.next().unwrap()).unwrap();
+                let fname = std::str::from_utf8(key_parts.next().unwrap()).unwrap();
```

Comment:
Not to tackle now, but those unwraps in the code are a bit worrying IMO. There's no validation of id but you're assuming a certain format. I thing we should be validating ids more explicitly so it's safer to make those assumptions

Reply:
I think we should tackle this now, for the iteration: it's pretty easy.

```rust
                if let Some(ftype) = key_parts.next() {
                    if let Some(fname) = key_parts.next() {
                        let ftype_str = std::str::from_utf8(ftype).unwrap();
                        let fname_str = std::str::from_utf8(fname).unwrap();
                        return ftype == field_type && fname == field_name;
                    }
                }
                false
```

I think the unwraps in the UTF8 decoding are fine, if those fail, the segment is corrupted and a noisy failure is in order.